aws/resource_aws_docdb_cluster_test.go

```diff
@@ -0,0 +1,635 @@
+				),
+			},
+		},
+	})
+}
+
+/// This is a regression test to make sure that we always cover the scenario as hightlighted in
+/// https://github.com/hashicorp/terraform/issues/11568
+func TestAccAWSDocDBCluster_missingUserNameCausesError(t *testing.T) {
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckDocDBClusterDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccDocDBClusterConfigWithoutUserNameAndPassword(acctest.RandInt()),
+				ExpectError: regexp.MustCompile(`required field is not set`),
+			},
+		},
+	})
+}
+
+func TestAccAWSDocDBCluster_updateTags(t *testing.T) {
+	var v docdb.DBCluster
+	ri := acctest.RandInt()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckDocDBClusterDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDocDBClusterConfig(ri),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDocDBClusterExists("aws_docdb_cluster.default", &v),
+					resource.TestCheckResourceAttr(
+						"aws_docdb_cluster.default", "tags.%", "1"),
+				),
+			},
+			{
+				Config: testAccDocDBClusterConfigUpdatedTags(ri),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDocDBClusterExists("aws_docdb_cluster.default", &v),
+					resource.TestCheckResourceAttr(
+						"aws_docdb_cluster.default", "tags.%", "2"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSDocDBCluster_updateCloudwatchLogsExports(t *testing.T) {
+	var v docdb.DBCluster
+	ri := acctest.RandInt()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckDocDBClusterDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDocDBClusterNoCloudwatchLogsConfig(ri),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDocDBClusterExists("aws_docdb_cluster.default", &v),
+				),
+			},
+			{
+				Config: testAccDocDBClusterConfig(ri),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDocDBClusterExists("aws_docdb_cluster.default", &v),
+					resource.TestCheckResourceAttr("aws_docdb_cluster.default",
+						"enabled_cloudwatch_logs_exports.0", "audit"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSDocDBCluster_kmsKey(t *testing.T) {
+	var v docdb.DBCluster
+	keyRegex := regexp.MustCompile("^arn:aws:kms:")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckDocDBClusterDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDocDBClusterConfig_kmsKey(acctest.RandInt()),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDocDBClusterExists("aws_docdb_cluster.default", &v),
+					resource.TestMatchResourceAttr(
```

Comment:
We can use `resource.TestCheckResourceAttrPair()` here to ensure the correct Terraform state attribute value instead of attempting to use regexp.

```go
resource.TestCheckResourceAttrPair("aws_docdb_cluster.default", "kms_key_id", "aws_kms_key.foo", "arn"),
```